ols/app/endpoints/ols.py

```diff
@@ -13,11 +13,21 @@
 from ols.src.llms.llm_loader import LLMConfigurationError, LLMLoader
 from ols.src.query_helpers.docs_summarizer import DocsSummarizer
 from ols.src.query_helpers.question_validator import QuestionValidator
+from ols.src.rag_index.index_loader import IndexLoader
 from ols.utils import config, suid
 
 logger = logging.getLogger(__name__)
 
-router = APIRouter(tags=["query"])
+_rag_index = None  # to avoid mypy error
+
+
+def _load_rag_index() -> None:
+    """Load vector index at start up."""
+    global _rag_index
+    _rag_index = IndexLoader(config.ols_config.reference_content).load_index()
+
+
+router = APIRouter(tags=["query"], on_startup=[_load_rag_index])
```

Comment:
my personal preference is to be consistent, that way we can keep all our "init" related logic in one place, otherwise we risk missing something during a refactor or when someone is trying to understand what happens at init time and doesn't realize there's two different sets of initialization logic/behaviors in play

i don't have a strong opinion on which approach to prefer, but i'd rather not use two different ones.